### PR TITLE
Generate .gitignore file when firebase init [functions]. Fix #272.

### LINKED ISSUE #272: Feature Request: add default .gitignore to ignore node_modules when running `firebase init functions`

### PR DESCRIPTION
**Pull request by yuchenshi**
### Description

This PR creates a `.gitignore` file when running `firebase init` and `firebase init functions`. Initializing the project will include a general `.gitignore` file which includes most unwanted logs and caches and `node_modules`. Initializing functions will create a `.gitignore` file if TypeScript is chosen, which ignores the compiled JS files. See #272 for the original discussion.

### Scenarios Tested

```
firebase init
firebase init functions  # Choose TypeScript here.
```

### CODE REVIEW

File: commands/init.js [hunk 1]
```diff
@@ -130,6 +132,12 @@ module.exports = new Command("init [feature]")
               "Press Space to select features, then Enter to confirm your choices.",
             choices: prompt.convertLabeledListChoices(choices),
           },
+          {
```
Comment (yuchenshi) [inline commands/init.js]: Alternatively, we can put the gitignore prompt first, which will remove the need of the `are you ready to proceed` hack on windows (see lines above). I'm not quite sure if that's a good idea though. Let me know if you'd prefer it that way.

**Comment by laurenzlong** [comment]
@mbleigh Do you have any thoughts on this?

**Comment by mbleigh** [comment]
I'm tempted to just silently create a `.gitignore` without any prompting.

1. Most developers use git-based workflows
2. A `.gitignore` file doesn't hurt anyone that doesn't use git

So the behavior would be "if a `.gitignore` file doesn't already exist in the project directory, create one."

**Comment by yuchenshi** [comment]
@mbleigh All right, I'll remove the prompt and make it write `.gitignore` if it doesn't exist.

**Comment by yuchenshi** [comment]
@laurenzlong @mbleigh Branch updated to only create `.gitignore` if not exists, without prompts. Please let me know if you have any other comments.

**Pull request merged**